src/awkward/_layout.py

```diff
@@ -250,7 +250,12 @@ def maybe_highlevel_to_lowlevel(obj):
         return obj
 
 
-def from_arraylib(array, regulararray, recordarray):
+def from_arraylib(
+    array,
+    regulararray,
+    recordarray,
+    primitive_policy: Literal["error", "promote", "pass-through"] = "promote",
```

Comment:
Why is the default `primitive_policy` to raise an error for some backends and to promote for other backends?

Reply:
Looking through the code, there are quite a lot of functions taking a primitive_policy optional argument, and the defaults vary between "promote", "error", and "pass-through." In _layout we have `HighLevelContext.unwrap` with `primitive_policy="promote"` in addition to the one I added this argument to, `from_arraylib`.

There are quite a lot of ways into from_arraylib. One common path is the high-level function [from_numpy](https://github.com/scikit-hep/awkward/blob/03f6169841ccd92714dd904e76e8531887dbe3c3/src/awkward/operations/ak_from_numpy.py#L55). But these already had a primitive_policy option which was being specified higher up the call stack, for example in [ak_fill_none](https://github.com/scikit-hep/awkward/blob/03f6169841ccd92714dd904e76e8531887dbe3c3/src/awkward/operations/ak_fill_none.py#L77-L84).

This argument allows us to add `primitive_policy="errer"` in, specifically, the [from_numpy](src/awkward/operations/ak_from_numpy.py) function.